### PR TITLE
planner: not push down mpp/join to TiFlash in some cases that TiFlash not supported the query

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

In some cases, TiFlash does not support MPP execution, so TiDB should disable generating mpp plan.
Issue Number: close pingcap/tics#1520 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- Disable mpp execution if the new collation is on.

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -557,6 +557,116 @@ func (s *testIntegrationSerialSuite) TestBroadcastJoin(c *C) {
...
+func (s *testIntegrationSerialSuite) TestJoinNotSupportedByTiFlash(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists table_1")
+	tk.MustExec("create table table_1(id int not null, bit_col bit(2) not null, datetime_col datetime not null)")
+	tk.MustExec("insert into table_1 values(1,b'1','2020-01-01 00:00:00'),(2,b'0','2020-01-01 00:00:00')")
+	tk.MustExec("analyze table table_1")
+
+	// Create virtual tiflash replica info.
+	dom := domain.GetDomain(tk.Se)
```
Comment (fzhedu) [inline planner/core/integration_test.go:569]: would here set tiflash replica like this way?
https://github.com/pingcap/tidb/pull/23056/files#diff-71a2dde4dc9f29476f96c621ad0c11f6fc3f7b45185e11cb16f44acf46954cfeR328

File: planner/core/exhaust_physical_plans.go [hunk 2]
```diff
@@ -1712,6 +1712,29 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 	return joins, true
 }
 
+func canExprsInJoinPushdown(p *LogicalJoin, storeType kv.StoreType) bool {
```
Comment (fzhedu) [inline planner/core/exhaust_physical_plans.go:1715]: Are there similar case for agg?

Comment (windtalker) [reply planner/core/exhaust_physical_plans.go:1715]: No

**Pull request merged**